### PR TITLE
LUN-44: Exercise header

### PR DESCRIPTION
**Pull request by SteffiStuffel**
This pull request belongs to an issue on our [bugtracker](https://issues.tuerantuer.org/secure/RapidBoard.jspa?rapidView=27&view=detail).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword _LUN_.

### CODE REVIEW

File: src/components/ExerciseHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import React, { useEffect, useLayoutEffect, useState } from 'react'
```
Comment (SteffiStuffel) [inline src/components/ExerciseHeader.tsx:1]: mostly from WriteExercise, which already had the Header and moved it here, to reuse.

**Review by zankerbe — APPROVED** [review]
Tested 👍

File: src/components/ExerciseHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+})
+
+interface ExerciseHeaderPropsType {
+  navigation: StackNavigationProp<RoutesParamsType, 'WordChoiceExercise' | 'ArticleChoiceExercise' | 'WriteExercise'>
+  route: RouteProp<RoutesParamsType, 'WordChoiceExercise' | 'ArticleChoiceExercise' | 'WriteExercise'>
+  currentWord: number
+  numberOfWords: number
+}
+
+function ExerciseHeader({ navigation, route, currentWord, numberOfWords }: ExerciseHeaderPropsType) {
```
Comment (klinzo) [inline src/components/ExerciseHeader.tsx:46]:
```suggestion
const ExerciseHeader = ({ navigation, route, currentWord, numberOfWords }: ExerciseHeaderPropsType) => {
```

**Review by klinzo — APPROVED** [review]

File: src/components/ExerciseHeader.tsx (hunk 2 above)
Comment (SteffiStuffel) [reply src/components/ExerciseHeader.tsx:46]: done

**Pull request merged**